src/axum_support.rs

```diff
@@ -0,0 +1,45 @@
+#[cfg(feature = "axum08")]
+#[cfg_attr(docsrs, doc(cfg(feature = "axum08")))]
+impl axum08::serve::Listener for crate::VsockListener {
+    type Io = crate::VsockStream;
+
+    type Addr = vsock::VsockAddr;
+
+    async fn accept(&mut self) -> (Self::Io, Self::Addr) {
+        loop {
+            match std::future::poll_fn(|cx| self.poll_accept(cx)).await {
+                Ok(tuple) => return tuple,
+                Err(err) => handle_accept_error(err).await,
+            }
+        }
+    }
+
+    fn local_addr(&self) -> std::io::Result<Self::Addr> {
+        self.local_addr()
+    }
+}
+
+#[cfg(feature = "axum08")]
+async fn handle_accept_error(err: std::io::Error) {
+    if matches!(
+        err.kind(),
+        std::io::ErrorKind::ConnectionRefused
+            | std::io::ErrorKind::ConnectionAborted
+            | std::io::ErrorKind::ConnectionReset
+    ) {
+        return;
```

Comment:
Shouldn't this also have some kind of exponential backoff or something, rather than continuously retrying as fast as possible?

Reply:
As far as I can see this will retry immediately in the case of these three types of errors, and sleep for a second for other errors. Am I missing something?